core/src/helpers.ts

```diff
@@ -28,6 +28,33 @@ const upperFirst = (value: string): string =>
 const lowerFirst = (value: string): string =>
   value.charAt(0).toLowerCase() + value.slice(1);
 
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+function deleteKeyFromObject(inputObject: any, keyToDelete: string) {
+  for (let [currentObjectKey, currentObjectValue] of Object.entries(
+    inputObject
+  )) {
+    if (currentObjectKey === keyToDelete) {
+      delete inputObject[keyToDelete];
+    } else if (Array.isArray(currentObjectValue)) {
+      deleteKeyFromObjectInArray(currentObjectValue, keyToDelete);
+    } else if (isObject(currentObjectValue)) {
+      deleteKeyFromObject(currentObjectValue, keyToDelete);
+    }
+  }
+}
+
+// eslint-disable-next-line  @typescript-eslint/no-explicit-any
```

Comment:
yeah, I know - any help/suggestions for typing this in a more specific manner would be appreciated.